src/decouplededitor.js

```diff
@@ -0,0 +1,249 @@
+	}
+
+	/**
+	 * Destroys the editor instance, releasing all resources used by it.
+	 *
+	 * @returns {Promise}
+	 */
+	destroy() {
+		this.ui.destroy();
+
+		return super.destroy();
+	}
+
+	/**
+	 * Creates a decoupled editor instance.
+	 *
+	 * Creating instance when using the {@glink builds/index CKEditor build}:
+	 *
+	 *		DecoupledEditor
+	 *			.create( '<p>Editor data</p>', {
+	 *				// The location of the toolbar in DOM.
+	 *				toolbarContainer: 'body div.toolbar-container',
+	 *
+	 *				// The location of the editable in DOM.
+	 *				editableContainer: 'body div.editable-container'
+	 *			} )
+	 *			.then( editor => {
+	 *				console.log( 'Editor was initialized', editor );
+	 *			} )
+	 *			.catch( err => {
+	 *				console.error( err.stack );
+	 *			} );
+	 *
+	 * Creating instance when using CKEditor from source (make sure to specify the list of plugins to load and the toolbar):
+	 *
+	 *		import DecoupledEditor from '@ckeditor/ckeditor5-editor-decoupled/src/decouplededitor';
+	 *		import Essentials from '@ckeditor/ckeditor5-essentials/src/essentials';
+	 *		import Bold from '@ckeditor/ckeditor5-basic-styles/src/bold';
+	 *		import Italic from '@ckeditor/ckeditor5-basic-styles/src/italic';
+	 *		import ...
+	 *
+	 *		DecoupledEditor
+	 *			.create( '<p>Editor data</p>', {
+	 *				plugins: [ Essentials, Bold, Italic, ... ],
+	 *				toolbar: [ 'bold', 'italic', ... ],
+	 *
+	 *				// The location of the toolbar in DOM.
+	 *				toolbarContainer: 'div.toolbar-container',
+	 *
+	 *				// The location of the editable in DOM.
+	 *				editableContainer: 'div.editable-container'
+	 *			} )
+	 *			.then( editor => {
+	 *				console.log( 'Editor was initialized', editor );
+	 *			} )
+	 *			.catch( err => {
+	 *				console.error( err.stack );
+	 *			} );
+	 *
+	 * **Note**: {@link module:core/editor/editorconfig~EditorConfig#toolbarContainer `config.toolbarContainer`} and
+	 * {@link module:core/editor/editorconfig~EditorConfig#editableContainer `config.editableContainer`} are optional. It is
+	 * possible to define the location of the UI elements manually once the editor is up and running:
+	 *
+	 *		DecoupledEditor
+	 *			.create( '<p>Editor data</p>' )
+	 *			.then( editor => {
+	 *				console.log( 'Editor was initialized', editor );
+	 *
+	 *				// Append the toolbar and editable straight into the <body> element.
+	 *				document.body.appendChild( editor.ui.view.toolbar.element );
+	 *				document.body.appendChild( editor.ui.view.editable.element );
+	 *			} )
+	 *			.catch( err => {
+	 *				console.error( err.stack );
+	 *			} );
+	 *
+	 * @param {String} data The data to be loaded into the editor.
+	 * @param {module:core/editor/editorconfig~EditorConfig} config The editor configuration.
+	 * @returns {Promise} A promise resolved once the editor is ready.
+	 * The promise returns the created {@link module:editor-decoupled/decouplededitor~DecoupledEditor} instance.
+	 */
+	static create( data, config ) {
```

Comment:
To avoid extending `EditorConfig` with additional properties, I think that we could define `DecoupledEditorConfig` as an extension of `EditorConfig` and add them there.